src/components/AppHero.vue

```diff
@@ -1,32 +1,50 @@
 <script setup lang="ts">
-import { onMounted, ref, computed } from 'vue';
+import { onMounted, ref, computed, watch } from 'vue';
+import { usePreferredDark } from '@vueuse/core';
 import { Theme } from '../types.d';
 
 const theme = ref<Theme | string>();
-const titleTheme = ref<string>();
+const isSystemDark = usePreferredDark();
 
-const systemTheme = computed(() => {
-  return window.matchMedia('(prefers-color-scheme: dark)').matches
-    ? Theme.Dark
-    : Theme.Light;
+watch(isSystemDark, _ => {
+  if (theme.value === Theme.System) {
+    updateTheme();
+  }
 });
 
-const themeSteps = computed(() => {
-  return systemTheme.value === Theme.Dark
+const themeSteps = computed<Array<string>>(() => {
+  return isSystemDark.value
     ? [Theme.System, Theme.Light, Theme.Dark]
     : [Theme.System, Theme.Dark, Theme.Light];
 });
 
-function getNextTheme() {
+function getNextTheme () {
   const themeIndex = themeSteps.value.findIndex(t => t === theme.value);
   const nextThemeIndex = (themeIndex + 1) % themeSteps.value.length;
 
   return themeSteps.value[nextThemeIndex];
 }
 
+const titleTheme = computed<string>(() => {
+  let title;
+  switch (getNextTheme()) {
+    case Theme.Dark:
+      title = 'Ubah ke Mode Gelap';
+      break;
+
+    case Theme.Light:
+      title = 'Ubah ke Mode Terang';
+      break;
+  
+    default:
+      title = 'Ubah ke Tema Sistem';
+      break;
+  }
+  return title;
```

Comment:
if we doesn't do any changes for title variable we can use return immediately to make it simpler.